### PR TITLE
add definition for publicKeyHex based on RFC4648

### PR DESCRIPTION
**Pull request by kdenhartog**
Signed-off-by: Kyle Den Hartog <kyle.denhartog@mattr.global>

### CODE REVIEW

**Comment by OR13** [comment]
We need this to be defined in a JSON-LD Context... since v3-unstable is currently blocked, how can we add things like this... @msporny ?

**Comment by msporny** [comment]
> since v3-unstable is currently blocked, how can we add things like this... @msporny ?

Why is v3-unstable blocked?

**Comment by kdenhartog** [comment]
Yup, that's the part I wasn't sure about if it should be added to v2 or v3. Will add to this PR before merging.

File: index.html [hunk 1]
```diff
@@ -1514,6 +1514,34 @@ <h3>publicKeyPem</h3>
         </pre>
       </section>
 
+      <section id="publicKeyHex" about="https://w3id.org/security#publicKeyHex"
+      typeof="rdf:Property">
+        <h3>publicKeyHex</h3>
+        <p>
+A public key Hex property is used to specify the hex-encoded version of the public key, based on <a href="https://tools.ietf.org/html/rfc4648#section-8">section 8 of rfc4648</a>. Hex encoding is also known as Base16 encoding.
```
Comment (msporny) [inline index.html:1521]:
```suggestion
The <code>publicKeyHex</code> property is used to specify the hex-encoded version of the public key, based on <a href="https://tools.ietf.org/html/rfc4648#section-8">section 8 of rfc4648</a>. Hex encoding is also known as Base16 encoding.
```

File: index.html [hunk 2]
```diff
@@ -1514,6 +1514,34 @@ <h3>publicKeyPem</h3>
...
+        <p>
+    The following example demonstrates the expression of a public key
+    in hex format.
+        </p>
+        <pre class="example prettyprint language-jsonld">
+{
+    "id": "did:example:123#hexKey",
+    "controller": "did:example:123",
+    "type": "EcdsaSecp256k1RecoveryMethod2020",
+    "publicKeyHex":"027560af3387d375e3342a6968179ef3c6d04f5d33b2b611cf326d4708badd7770"
```
Comment (msporny) [inline index.html:1540]: This example is wrong -- it uses lowercase letters, which the IETF RFC you point to does not. :) ... decisions, decisions. :P 

You should just do what the referred to spec says (use uppercase). There will, inevitably, be people that want to do both, so it'll be up to you to fight that battle and not give in. The easy path will be to support upper and lowercase, and if you do that, you'll destroy the deterministic sorting provided by base32hex.

Comment (kdenhartog) [reply index.html:1540]: Ohhh good catch, let's define it properly here and when I catch people are doing it wrong I'll refer them here to point them in this direction.

Comment (OR13) [reply index.html]: ahhhh! yes this is a great catch... I gotten beaten up by this issue in my NFC implementation.

**Comment by OR13** [comment]
> > since v3-unstable is currently blocked, how can we add things like this... @msporny ?
> 
> Why is v3-unstable blocked?

https://github.com/w3c-ccg/security-vocab/pull/35

I suppose we can define the file again in this PR.

**Comment by kdenhartog** [comment]
> > > since v3-unstable is currently blocked, how can we add things like this... @msporny ?
> > 
> > 
> > Why is v3-unstable blocked?
> 
> #35
> 
> I suppose we can define the file again in this PR.

Yeah I can do it that way and then we'll just need to resolve the merge conflicts on #35 after. At least then we get v3 started as well.

**Pull request merged**